src/polygon-layer.ts

```diff
@@ -0,0 +1,297 @@
+import {
+  CompositeLayer,
+  CompositeLayerProps,
+  DefaultProps,
+  Layer,
+  LayersList,
+  GetPickingInfoParams,
+  assert,
+} from "@deck.gl/core/typed";
+import { PolygonLayer } from "@deck.gl/layers/typed";
```

Comment:
We should try rendering the SolidPolygonLayer and PathLayer ourselves instead of using the upstream PolygonLayer. 

E.g. the performance guide mentions:

> Note that external attributes only work with primitive layers, not composite layers, because composite layers often need to preprocess the data before passing it to the sub layers